### ISSUE TITLE
Adding tabs

### ISSUE THREAD

**Issue by glvr182** [awesome-gocui/gocui#10]
**Describe the feature you'd like**
Sometime there is just not enough room in the terminal, or two views are  
closely related. This calls for a new thing called a "tab". Just like with your browser, a tab holds different information but does not require you to start up a new window.

**Describe alternatives you've considered**
Manual tabbing, but it is a pain for the developer to create this each and every time.

**Additional context**
The title of the view would look like this: ```─Tab one / Tab two─```. and the current tab would be highlighted.

**Comment by mjarkk** [comment]
The idea of tabs is amazing thought `─Tab one / Tab two─` looks a bit off to me.  
I'm also wondering how this would look on small screens when having more than 2 tabs: `─Tab one/T/T/T─`.

**Comment by glvr182** [comment]
Agreed, I still have to find a good way of displaying tabs. This was my first thought. I think when terminals are too small to display a title we should cut it per tab. So lets say it is first `First / Second` it would go to  
   `F / S`

**Comment by vladkras** [comment]
Or maybe 
```┌─| First | Second |──```

```┌─ First | Second ──```

**Comment by mjarkk** [comment]
@vladkras I think it's a bad idea to use `|` because it's the same character as the box frame.  
I think this could be confusing when the terminal gets too small or overlap because the `|` might look like a border.

**Comment by glvr182** [comment]
Picking this up again after #41  
Will integrate the following commits:  
https://github.com/jesseduffield/gocui/commit/619845fffd4d94c2e248f04bcdeaa2669ae24698  
https://github.com/jesseduffield/gocui/commit/32197598edc52af42d76b8cba78ed788074b4647  
https://github.com/jesseduffield/gocui/commit/117e4fb79d29e3908cb7a7bed5e991a321fde7fd
